website/client/src/components/auth/socialAuth/apple.vue

```diff
@@ -0,0 +1,109 @@
+<template>
```

Comment:
Great start with adding some much need abstraction layers!

However, looking at the level of duplication between this one and the facebook and google one, perhaps they can be abstracted further.

The only things that are different in the three components are `networkName`, `icon` and the `socialAuth` method. Ideally you could create the three components by only supplying those three differences.

Edit: Now see there's a little extra markup in the apple-icon as well. Perhaps we can add that in as an optional parameter or something like that.

Reply:
i abstracted multi social auth component into one, and indeed they repeat lots of codes